src/main/java/io/github/genomicdatainfrastructure/discovery/datasets/infrastructure/beacon/persistence/BeaconIndividualsRequestMapper.java

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package io.github.genomicdatainfrastructure.discovery.datasets.infrastructure.beacon.persistence;
+
+import static java.util.Optional.ofNullable;
+import static java.util.stream.Collectors.toMap;
+
+import io.github.genomicdatainfrastructure.discovery.model.DatasetSearchQuery;
+import io.github.genomicdatainfrastructure.discovery.model.DatasetSearchQueryFacet;
+import io.github.genomicdatainfrastructure.discovery.model.FilterType;
+import io.github.genomicdatainfrastructure.discovery.model.QueryEntry;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconIndividualsRequest;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconIndividualsRequestMeta;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconIndividualsRequestQuery;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconIndividualsRequestQueryFilter;
+import io.github.genomicdatainfrastructure.discovery.remote.beacon.model.BeaconIndividualsRequestQueryPagination;
+import jakarta.ws.rs.BadRequestException;
+import lombok.experimental.UtilityClass;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+@UtilityClass
+public class BeaconIndividualsRequestMapper {
+
+    private static final String BEACON_FACET_GROUP = "beacon";
+    private static final String SCOPE = "individual";
+    private static final String INCLUDE_RESULTSET_RESPONSES = "HIT";
+    private static final String REQUESTED_GRANULARITY = "record";
+
+    public BeaconIndividualsRequest from(
+            DatasetSearchQuery query
+    ) {
+        var nonNullFacets = ofNullable(query)
+                .map(DatasetSearchQuery::getFacets)
+                .filter(ObjectUtils::isNotEmpty)
+                .orElseGet(List::of);
+
+        var beaconFilters = nonNullFacets.stream()
+                .filter(it -> BEACON_FACET_GROUP.equals(it.getSource()))
+                .map(BeaconIndividualsRequestMapper::buildBeaconFilter)
+                .filter(Objects::nonNull)
```

Comment:
important: can you please remove this filter?

A mapper should always receive an object and retrieve another.